src/common/weighted_shuffle.h

```diff
@@ -0,0 +1,41 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_WEIGHTED_SHUFFLE_H
+#define CEPH_WEIGHTED_SHUFFLE_H
+
+#include <vector>
+#include <random>
+#include <algorithm>
```

Comment:
nit, reorder the `#include`s alphabetically . see https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes